### DISCUSSION TITLE
Web service: 433 failed to response

### DISCUSSION THREAD

**Discussion by hjp37** [SalesforceCommerceCloud/sfcc-ci#535] [Team Posts]
Hi guys, 

I am writing to see if anyone has experienced, or knows how to handle the below problem.
There is this inventory api we are using to confirm allocation and this api randomly fails to respond. 
It's been driving me nuts and my knowledge might be lacking to fix this :(. 
The third party says they never got the request call, so it must be the internal problem.
But it works most of the time, so I just can't find the root cause to fix this. 

From the service log, it's printed as below:

>  ERROR PipelineCallServlet|22456|Sites-xxx-Site|COSummary-Submit|PipelineCall|1DCGSXpItV custom.service.xxx_http_service.HEAD []  service=xxx_http_service status=ERROR errorCode=0 errorMessage=NoHttpResponseException:xxx.xxxxx.xx:443 failed to respond

Any help would be appreciated. 
Thank you!

Regards, 
Hyejin

**Comment by pravngaur** [comment]
Hi,

I just googled for such errors and figured out that such errors happens at
client side only, requests does not reach server at all.

Few resources :

https://github.com/DigitalPebble/storm-crawler/issues/405

https://stackoverflow.com/questions/26111331/org-apache-http-nohttpresponseexception-xx-xx-xx-xx443-failed-to-respond

Regarding your specific issue, at what frequency you are calling this
service? It might be caused by the exhausted resources /threads.

Is it happening on production or on your development environment?

If on development, try reducing the frequency and see if it goes away, if
it does then loop in the platform support team and ensure that you would
not face the same issue on production at higher frequency.

If it is happening on prod then definitely loop in platform support team.

I don't think this is a application layer issue but has to do something
with the resource pooling at server.

Thanks.

**Comment by tparikh2** [comment]
Hi,

Please check your service configuration.

What is the time out value?

If circuit breaker enable then please check the configuration.

Try to increase the calls per second in circuit breaker and check.

Cheers,
Tushar

**Comment by hjp37** [comment]
@pravngaur 
We are exceeding number of web service calls in one request (but the quota limit has been temporarily removed by the support team), so that might be one of the causes. 
Thank you for your comment and links! I will look more into it :).

**Comment by pravngaur** [comment]
Great, i believe this should be the root cause. Work with platform team.
Cheers.

**Comment by hjp37** [comment]
@tparikh2 
We've set 3,000ms for the timeout value, and set Max Circuit Break Calls to 3, and Circuit Breaker Interval to 1,000ms. The circuit breaker call is always bit confusing so I am re-checking the documentation. Thank you for your comment Tushar!
